src/pushsource/_impl/backend/staged/staged_cloud.py

```diff
@@ -0,0 +1,128 @@
+    def __build_ami_push_item(self, resources, src, origin, name):
+        build_resources = resources.get("build")
+        release_resources = resources.get("release", {})
+        build_info = KojiBuildInfo(
+            name=build_resources.get("name"),
+            version=build_resources.get("version"),
+            release=build_resources.get("respin"),
+        )
+        image_kwargs = {
+            "name": name,
+            "src": src,
+            "build_info": build_info,
+            "origin": origin,
+        }
+
+        image_kwargs.update(
+            {
+                "boot_mode": (
+                    BootMode(resources.get("boot_mode"))
+                    if resources.get("boot_mode")
+                    else None
+                )
+            }
+        )
+
+        release_required = ["product", "date", "arch", "respin"]
+        if all(x in release_resources.keys() for x in release_required):
+            release_attrs = [
+                "product",
+                "date",
+                "arch",
+                "respin",
+                "version",
+                "base_product",
+                "base_version",
+                "variant",
+                "type",
+            ]
+            release_kwargs = {}
+            for key in release_attrs:
+                release_kwargs[key] = release_resources.get(key)
+
+            image_kwargs["release"] = AmiRelease(**release_kwargs)
+
+        image_attrs = [
+            "dest",
+            "type",
+            "region",
+            "virtualization",
+            "volume",
+            "root_device",
+            "description",
+            "sriov_net_support",
+            "ena_support",
+            "uefi_support",
+            "public_image",
+            "release_notes",
+            "usage_instructions",
+            "recommended_instance_type",
+            "marketplace_entity_type",
+            "image_id",
+            "scanning_port",
+            "user_name",
+            "version_title",
+            "security_groups",
+            "access_endpoint_url",
+        ]
+
+        for key in image_attrs:
+            if key in resources:
+                image_kwargs[key] = resources.get(key)
+
+        return AmiPushItem(**image_kwargs)
+
+    def __build_azure_push_item(self, resources, src, origin, name):
+        build_resources = resources.get("build")
+        build_info = KojiBuildInfo(
+            name=build_resources.get("name"),
+            version=build_resources.get("version"),
+            release=build_resources.get("respin"),
+        )
+        image_kwargs = {
+            "name": name,
+            "src": src,
+            "description": resources.get("description"),
+            "build_info": build_info,
+            "origin": origin,
+        }
+        return VHDPushItem(**image_kwargs)
+
+    @handles_type("CLOUD_IMAGES")
+    def __cloud_push_item(self, leafdir, _, entry):
+        with open(entry.path, "rt") as fh:
```

Comment:
This does not seem right; how do you know this is only opening the `resources.yaml`?  Isn't this code going to be executed on every file within the `CLOUD_IMAGES` directory?

The tests are probably only able to pass because every sample image you added is an empty file, and `yaml.safe_load` will happily accept an empty string and return `None`.  I think if you added any non-empty VMI to the test data then the tests would crash, detecting the bug here.  So can you please do that?  (doesn't have to be a real VMI.)